src/server/scripts/Commands/cs_ahbot.cpp

```diff
@@ -146,12 +146,28 @@ class ahbot_commandscript : public CommandScript
     static bool HandleAHBotRebuildCommand(ChatHandler* /*handler*/, const char* args)
     {
         char* arg = strtok((char*)args, " ");
+        char* ahMapIdStr = strtok(NULL, " ");
+        uint32 ahMapID = MAX_AUCTION_HOUSE_TYPE;
 
         bool all = false;
         if (arg && strcmp(arg, "all") == 0)
             all = true;
 
-        sAuctionBot->Rebuild(all);
+        if (ahMapIdStr)
+        {
+            // NOTE: Should use _stricmp() for case insenstive, stricmp() is deprecated
```

Comment:
aren't all these if/else just a switch on ahMapIdStr[0] ? No need for any stricmp